Engine/source/platform/platformNet.cpp

```diff
@@ -823,13 +823,16 @@ NetSocket Net::openConnectTo(const char *addressString)
       if (socketFd != InvalidSocketHandle)
       {
          setBlocking(handleFd, false);
-         if (::connect(socketFd, (struct sockaddr *)&ipAddr, sizeof(ipAddr)) == -1 &&
-            errno != EINPROGRESS)
+         if (::connect(socketFd, (struct sockaddr *)&ipAddr, sizeof(ipAddr)) == -1)
```

Comment:
From what i've read, there are two possible states here for non-blocking sockets:

returns 0 (e.g. for localhost)
returns -1, errno is EINPROGRESS (e.g. for remote connections)

Probably should be used in this case. Naturally though this seems to be sporadically documented so I cant be 100% sure this is true for every single sockets impl....